in/kafkamdm/kafkamdm.go

```diff
@@ -140,11 +154,13 @@ func New(stats met.Backend) *KafkaMdm {
 
 func (k *KafkaMdm) Start(metrics mdata.Metrics, metricIndex idx.MetricIndex, usg *usage.Usage) {
 	k.In = in.New(metrics, metricIndex, usg, "kafka-mdm", k.stats)
+	var err error
 	for _, topic := range topics {
-		// get partitions.
-		partitions, err := k.consumer.Partitions(topic)
-		if err != nil {
-			log.Fatal(4, "kafka-mdm: Faild to get partitions for topic %s. %s", topic, err)
+		if partitionStr == "*" {
```

Comment:
even if partitionStr is != "*" we should fetch the list of partitions for the topic so that we can validate the passed partition list